src/test/xmeta/test_symbol_table.cpp

```diff
@@ -517,3 +515,377 @@ TEST_CASE("Resolving types across namespaces test")
+    REQUIRE((method1_return_type.is_resolved() && std::get<simple_type>(method1_return_type.get_resolved_target()) == simple_type::Int32));
+    auto const& method1_formal_parameters = method1->get_formal_parameters();
+    REQUIRE(method1_formal_parameters.size() == 2);
+    REQUIRE(method1_formal_parameters[0].get_id() == "i");
+    REQUIRE(std::get<simple_type>(method1_formal_parameters[0].get_type().get_semantic().get_resolved_target()) == simple_type::Int32);
+    REQUIRE(method1_formal_parameters[1].get_id() == "d");
+    REQUIRE(std::get<simple_type>(method1_formal_parameters[1].get_type().get_semantic().get_resolved_target()) == simple_type::Int32);
+}
+
+TEST_CASE("Resolving interface method type ref test")
+{
+    std::istringstream test_idl{ R"(
+        namespace N
+        {
+            struct S1
+            {
+            };
+
+            interface IControl
+            {
+                E1 Draw(S1 p1, M.S2 p2);
+            }
+
+            enum E1
+            {
+            }
+        }
+        namespace M
+        {
+            struct S2
+            {
+            };
+        }
+    )" };
+
+    xmeta_idl_reader reader{ "" };
+    reader.read(test_idl);
+    REQUIRE(reader.get_num_syntax_errors() == 0);
+
+    auto const& namespaces = reader.get_namespaces();
+    auto const& it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto const& ns_bodies = it->second->get_namespace_bodies();
+    REQUIRE(ns_bodies.size() == 1);
+    auto const& interfaces = ns_bodies[0]->get_interfaces();
+    REQUIRE(interfaces.size() == 1);
+
+    REQUIRE(interfaces.find("IControl") != interfaces.end());
+    auto const& model = interfaces.at("IControl");
+
+    auto const& methods = model->get_methods();
+    REQUIRE(methods[0]->get_id() == "Draw");
+
+    auto const& return_type = methods[0]->get_return_type()->get_semantic();
+    REQUIRE(return_type.is_resolved());
+    REQUIRE(std::get<std::shared_ptr<enum_model>>(return_type.get_resolved_target())->get_id() == "E1");
+
+    auto const& params = methods[0]->get_formal_parameters();
+    auto const& param_type = params[0].get_type().get_semantic();
+    auto const& properties = model->get_properties();
+    {
+        REQUIRE(params[0].get_id() == "p1");
+        auto const& param_type = params[0].get_type().get_semantic();
+        REQUIRE(param_type.is_resolved());
+        REQUIRE(std::get<std::shared_ptr<struct_model>>(param_type.get_resolved_target())->get_id() == "S1");
+    }
+    {
+        REQUIRE(params[1].get_id() == "p2");
+        auto const& param_type = params[1].get_type().get_semantic();
+        REQUIRE(param_type.is_resolved());
+        REQUIRE(std::get<std::shared_ptr<struct_model>>(param_type.get_resolved_target())->get_id() == "S2");
+    }
+}
+
+TEST_CASE("Interface property test")
+{
+    std::istringstream test_idl{ R"(
+        namespace N
+        {
+            interface IControl
+            {
+                Int32 property1 { get; set; };
```

Comment:
I see validation that there are three methods, and two properties. However, the model should also be linking the property to its getter and setter (or vice versa) - I don't see that relationship being checked.